generators/entity/index.js

```diff
@@ -222,17 +222,20 @@ module.exports = class extends BaseGenerator {
                 const prodDatabaseType = context.prodDatabaseType;
                 const entityTableName = context.entityTableName;
                 const jhiTablePrefix = context.jhiTablePrefix;
+                const instructions = `You can specify a different table name in your JDL file or change it in .jhipster/${context.name}.json file and then run again 'jhipster entity ${context.name}.'`;
+
                 if (!(/^([a-zA-Z0-9_]*)$/.test(entityTableName))) {
-                    this.error(chalk.red('The table name cannot contain special characters'));
+                    this.error(chalk.red(`The table name cannot contain special characters.\n${instructions}`));
                 } else if (entityTableName === '') {
                     this.error(chalk.red('The table name cannot be empty'));
                 } else if (jhiCore.isReservedTableName(entityTableName, prodDatabaseType)) {
-                    this.warning(chalk.red(`The table name cannot contain the '${entityTableName.toUpperCase()}' reserved keyword, so it will be prefixed with '${jhiTablePrefix}_'`));
+                    this.warning(chalk.red(`The table name cannot contain the '${entityTableName.toUpperCase()}' reserved keyword, so it will be prefixed with '${jhiTablePrefix}_'.\n${instructions}`));
                     this.entityTableName = `${jhiTablePrefix}_${entityTableName}`;
+                    context.entityTableName = this.entityTableName;
```

Comment:
I think it was my bad, when i made changes to use context instead of this, i might have missed this line. So I think the `this.entityTableName` assignment is not needed here only the context assignment you added would be sufficient. Can you plz check?

Reply:
OK I'll try